_posts/2020-07-09-implementing-retry-with-resilience4j.md

```diff
@@ -134,7 +134,7 @@ Flight search successful
 [Flight{flightNumber='XY 765', flightDate='07/31/2020', from='NYC', to='LAX'}, ...]
 ```
 
-Now, suppose we're calling `FlightSearchService.searchFlightsThrowingException()` instead, which can throw a checked `Exception`. Then we would get a compiler error on this line:
+Now, suppose we want to retry for both checked and unchecked exceptions. Let's say we're calling `FlightSearchService.searchFlightsThrowingException()` which can throw a checked `Exception`. Since a `Supplier` cannot throw a checked exception, we would get a compiler error on this line:
```

Comment:
Here it says that checked exceptions are being retried, but they aren't retried with `decorateCheckedSupplier()`.

Reply:
Checked exceptions are retried when we use decorateCheckedSupplier() @thombergs . Which example did you play around with? Please see basicExample_ServiceThrowingException_v2(). You will see an output like this:

Searching for flights; current time = 14:28:06 358
Searching for flights; current time = 14:28:06 870
Searching for flights; current time = 14:28:07 370

The reason we have a catch block in application code is that the framework can throw the exception after the final retry. The signature of CheckedFunction0.apply is:

R apply() throws Throwable;

That's why the catch block (in the article) says: " // handle exception that can occur after retries are exhausted"